src/kudu/security/tls_handshake.cc

```diff
@@ -79,6 +79,57 @@ void TlsHandshake::SetSSLVerify() {
   SSL_set_verify(ssl_.get(), ssl_mode, /* callback = */nullptr);
 }
 
+// Perform a normal TLS handshake
```

Comment:
Any reason why we created brand new functions for this?

We had ClientNegotiation::HandleTlsHandshake(const NegotiatePB& response) {
in which we seem to be doing 2 things.

1. tls_context_->InitiateHandshake
2. tls_handshake_.Continue()

It seems we are redoing the entire SSL code path which kudu already has.
Except that it was prefixed with a custom negotiation.